### PR TITLE
Add status management and enhance deletion

### LINKED ISSUE #5: Add Status fields management
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

Status management: 

We need to manage the status field to report the progress of secrets and configmaps copying.
The status could have two fields members and phase

```
status:
  members:
  - secret1: Completed
  - secret2: NotFound
  - configmap1: Updating
 Phase: Completed  
 ```

 

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by horis233**
**What this PR does / why we need it**:

This PR adds status management for the CR and when the original secret/configmap are deleted, the copied secret/configmap will be deleted as well.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/controller/secretshare/secretshare_controller.go [hunk 1]
```diff
@@ -130,36 +136,59 @@ func (r *ReconcileSecretShare) copySecret(instance *ibmcpcsv1.SecretShare) bool
...
-		}
-		for _, ns := range secretShare.Sharewith {
-			if err := r.ensureNs(ns.Namespace); err != nil {
+			if errors.IsNotFound(err) && instance.CheckStatus(ns+"/"+secretName, ibmcpcsv1.Running) {
+				if r.deleteCopiedSecret(secretName, secretShare) {
+					instance.UpdateStatus(ns+"/"+secretName, ibmcpcsv1.Failed)
+					requeue = true
+					continue
+				}
+				instance.RemoveStatus(ns + "/" + secretName)
```
Comment (Daniel-Fan) [inline pkg/controller/secretshare/secretshare_controller.go:149]: May I ask that why we do need to remove status when there is no err found in errors and the previous status is running?
Whatever we deleted copied secret successfully or not.

Comment (horis233) [reply pkg/controller/secretshare/secretshare_controller.go:149]: The Controller will remove the status if the secret/configmap is not found and its status is running, which means the original secret/configmap has been deleted. The secretshare operator will delete the copied secret/configmap and clean up the status.

**Comment by Daniel-Fan** [comment]
\lgtm

**Pull request merged**